src/NMEA_build.cpp

```diff
@@ -565,7 +565,13 @@ char *Adafruit_GPS::build(char *nmea, const char *thisSource,
 
   addChecksum(nmea); // Successful completion
   if (!noCRLF) { // Add Carriage Return and Line Feed to comply with NMEA-183
-    sprintf(nmea, "%s\r\n", nmea);
+    size_t len = strlen(nmea);
+    char *newStr =
+        (char *)malloc(len + 3); // +2 for \r\n, +1 for null terminator
+    if (newStr) {
+      strcpy(newStr, nmea);   // Copy original string
+      strcat(newStr, "\r\n"); // Append \r\n
```

Comment:
newstr is never used after this and we return nmea so this code does not actually append \r\n to nmea?

Reply:
Thanks, I'll take another look at the code.